crates/rpc-types/src/eth/log.rs

```diff
@@ -26,33 +24,67 @@ pub struct Log {
     pub removed: bool,
 }
 
-impl TryFrom<Log> for alloy_primitives::LogData {
-    type Error = LogError;
+impl<T> Log<T>
+where
+    for<'a> &'a T: Into<alloy_primitives::LogData>,
+{
+    /// Reserialize the data.
+    pub fn reserialize(&self) -> Log<alloy_primitives::LogData> {
```

Comment:
I see, this can be used to convert anything other than LogData in the default `Log`

could add this as context to "deserialize the data into the default Log type"

if we have &self, we can also add an `Into<Log>` either by impl Into trait (perhaps not possible) or `into_reserialize`